commit 5a27136c9df7bb9e920c9f0bd91ed2af7b54c778
Author: Michel Hermier <hermier@frugalware.org>
Date:   Mon Jul 5 13:50:37 2010 +0200

    scripts/makepkg: Disable ccache when /var/cache/ccache is not writable.
    
    * With the recent cache update, it now don't die silently when the cache
      is writable. Don't export ccache environment in this case (so that
      ccache is disabled).

diff --git a/scripts/makepkg b/scripts/makepkg
index 5988ccf..c177dd4 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1537,13 +1537,14 @@ fi
 
 # use ccache if it's available
 if [ "$NOCCACHE" = "0" -a -d /usr/lib/ccache/bin ]; then
-	export PATH=/usr/lib/ccache/bin:$PATH
-	FCC="ccache $FCC"
-	install -d -m 2775 /var/cache/ccache/$pkgname
-	export CCACHE_DIR=/var/cache/ccache/$pkgname
-	export CCACHE_NOLINK=1
-	export CCACHE_UMASK=002
-	if [ ! -w /var/cache/ccache/ ]; then
+	if [ -w /var/cache/ccache/ ]; then
+		export PATH=/usr/lib/ccache/bin:$PATH
+		FCC="ccache $FCC"
+		install -d -m 2775 /var/cache/ccache/$pkgname
+		export CCACHE_DIR=/var/cache/ccache/$pkgname
+		export CCACHE_NOLINK=1
+		export CCACHE_UMASK=002
+	else
 		if [ "$INCHROOT" != "1" ]; then
 			error "You installed ccache but you are not in the ccache group."
 			plain "Either add yourserlf to the ccache group or"
